Sources/MongoKitten/AggregateStage.swift

```diff
@@ -206,4 +206,72 @@ public struct AggregateBuilderStage {
+    ///   - includeLocs: Optional. This specifies the output field that identifies the location used to calculate the distance. This option is useful when a location field contains multiple locations. To specify a field within an embedded document, use dot notation.
+    ///   - uniqueDocs: Optional. If this value is true, the query returns a matching document once, even if more than one of the document’s location fields match the query.
+    ///   - minDistance: Optional. The minimum distance from the center point that the documents can be. MongoDB limits the results to those documents that fall outside the specified distance from the center point.
+    ///   - key: Specify the geospatial indexed field to use when calculating the distance. If your collection has multiple 2d and/or multiple 2dsphere indexes, you must use the key option to specify the indexed field path to use. Specify Which Geospatial Index to Use provides a full example. If there is more than one 2d index or more than one 2dsphere index and you do not specify a key, MongoDB will return an error. If you do not specify the key, and you have at most only one 2d index and/or only one 2dsphere index, MongoDB looks first for a 2d index to use. If a 2d index does not exists, then MongoDB looks for a 2dsphere index to use.
+    public static func geoNear(
+        useLegacy: Bool = false,
+        coordinates: [Double],
+        distanceField: String,
+        spherical: Bool = false,
+        maxDistance: Int? = nil,
+        query: Document? = nil,
+        distanceMultiplier: Int? = nil,
+        includeLocs: String? = nil,
+        uniqueDocs: Bool? = nil,
+        minDistance: Int? = nil,
+        key: String? = nil
+    ) -> AggregateBuilderStage {
+        var geoNear: Document = ["distanceField": distanceField,
+                                 "spherical": spherical]
+        
+        if useLegacy {
+            geoNear["near"] = coordinates.makePrimitive()
+        } else {
+            geoNear["near"] = ["type": "Point", "coordinates": coordinates] as Document
+        }
+                
+        if let maxDistance = maxDistance {
+            geoNear["maxDistance"] = maxDistance
+        }
+        
+        if let query = query {
+            geoNear["query"] = query
+        }
+        
+        if let distanceMultiplier = distanceMultiplier {
+            geoNear["distanceMultiplier"] = distanceMultiplier
+        }
+        
+        if let includeLocs = includeLocs {
+            geoNear["includeLocs"] = includeLocs
+        }
+        
+        if let uniqueDocs = uniqueDocs {
+            geoNear["uniqueDocs"] = uniqueDocs
+        }
+        
+        if let minDistance = minDistance {
+            geoNear["minDistance"] = minDistance
+        }
+        
+        if let key = key {
```

Comment:
You can assign a `String?` directly to a Document. If it's `nil` it won't be inserted.

Reply:
Should simplify this code a LOT

Reply:
Ah that’s good to know. Thanks for the tip. Will update.

Reply:
Resolved